src/kirby/components/shared/resize-observer/resize-observer.service.ts

```diff
@@ -0,0 +1,67 @@
+import { ElementRef, Injectable, OnDestroy, NgZone } from '@angular/core';
+
+import { ResizeObserver } from './types/resize-observer';
+import { ResizeObserverEntry } from './types/resize-observer-entry';
+import { ResizeObserverFactory } from './resize-observer.factory';
+
+@Injectable({ providedIn: 'root' })
+export class ResizeObserverService implements OnDestroy {
+  // IMPORTANT: Keeps track of the existing ResizeObserver so it can be reused for performance.
+  // See: https://groups.google.com/a/chromium.org/forum/#!msg/blink-dev/z6ienONUb5A/F5-VcUZtBAAJ
+  private observer: ResizeObserver | null;
+  private observedElements = new WeakMap<Element, (entry: ResizeObserverEntry) => void>();
+
+  constructor(private _resizeObserverFactory: ResizeObserverFactory, private zone: NgZone) {}
+
+  observe(elementOrRef: Element | ElementRef<Element>, action: (entry: ResizeObserverEntry) => void): Element {
+    if (this.observer === undefined) {
+      this.observer = this._resizeObserverFactory.create((entries) => this.handleResize(entries));
```

Comment:
Could the assignment not be moved to the constructor. Then the observe function be changed to:
 
```suggestion
    const element = elementOrRef instanceof ElementRef ? elementOrRef.nativeElement : elementOrRef;

    if (!this.observer) {
      return element;
    }

    if (!this.observedElements.has(element)) {
      // IMPORTANT: Has to be run outside the Angular zone, for it to work with ResizeObserver polyfill:
      this.zone.runOutsideAngular(() => {
        this.observer.observe(element);
      });
      this.observedElements.set(element, action);
    }
    return element;
```

Would that not be a bit more readable?

Reply:
Yes, but then we create the `ResizeObserver` before it's actually needed (that is, when `observe` is called). All though that might be a bit theoretical since we only create the entire service if it's needed. I've changed it + removed the return value as we didn't use it anywhere.

Reply:
Cross-browser support is not very good for `ResizeObserver` even with polyfills, can we take another path?

Reply:
Not really sure what you mean by this? Have you looked at the polyfill implementation or are you basing your comment on caniuse.com alone? Have you compared it with the supported browserlist for DRB?